constraint/pkg/apis/templates/v1/constrainttemplate_types.go

```diff
@@ -34,7 +34,8 @@ type CRD struct {
 }
 
 type CRDSpec struct {
-	Names      Names       `json:"names,omitempty"`
+	Names Names `json:"names,omitempty"`
+	// +kubebuilder:default={legacySchema: false}
```

Comment:
This strikes me as the case of `Validation` being a `nil` pointer, which then should be defaulted to:

``` { legacySchema: false }```

But that seems different than the test cases that were added in `constrainttemplate_types_test.go`, which are about `Validation.OpenAPIV3Schema` being `nil`.

Can you elaborate?

Reply:
I see now that the two defaults serve two different cases: `Validation: nil` and `Validation: &Validation{}`

Reply:
If `validation` is nil, then K8s defaulting doesn't go further, which means the default value for `legacySchema` is never applied.

The test validates that `legacySchema` behaves properly when `openAPIV3Schema` is nil. The extra defaulting makes it so that that logic actually gets invoked.